assets/src/edit-story/components/inspector/document/publish/index.js

```diff
@@ -99,6 +113,31 @@ function PublishPanel() {
     [updateStory]
   );
 
+  const generateCanvas = useCallback(async () => {
+    try {
+      const canvas = await html2canvas(fullbleedContainer);
+      const blob = await new Promise((resolve, reject) =>
+        canvas.toBlob(
+          (value) => (value ? resolve(value) : reject(new Error())),
+          'image/jpeg'
+        )
+      );
+      const { id, media_details, source_url } = await uploadFile(
+        new File([blob], 'cover-image.jpg', { type: 'image/jpeg' })
+      );
```

Comment:
The `uploadFile` hook is more direct than `uploadMedia` without having to merge current assets. If we want this generated cover image to show up in their media gallery then either we could refresh it or use the `uploadMedia` hook but I didn't want to over complicate the logic here.

Reply:
Yeah I don't think we'd want this in the media library

Reply:
It gets added on refresh by nature of it being a piece of media.

Reply:
This relates to Jonny's comment here. We do have some functionality for this in place already, but might need to slightly tweak it in order to be able to handle this type of image correctly.